### ISSUE TITLE
Instance Segmentation Loss

### ISSUE THREAD

**Issue by npmhung** [ayushjain1144/odin#38]
Thank you for the great work.

I have a question regarding the loss for instance segmentation. I observe that you do the Hungarian matching independently on each view as shown  [here](https://github.com/ayushjain1144/odin/blob/cc60d01418b03f4d8db84429307ff6ae7d2622d3/odin/modeling/matcher.py#L233) . Therefore,  each query can be matched to a different instance in each view. How do you prevent that from happening?

**Comment by ayushjain1144** [comment]
Hi, It does not happen per view. It happens on the whole RGB-D point cloud

**Comment by npmhung** [comment]
@ayushjain1144 Hi, could you further explain how does that happen or point me to the code that suppor that logic? 
I checked the Matcher. The input shape is  [batch_size, num_queries, num_classes] and  [batch_size, num_queries, H_pred, W_pred] for the pred_logits and pred_masks. I assume that ```batch_size = real_batchsize * view```.  However, I observe that you calculate the matching per sample/view and do not see where all the views are considered.

**Comment by ayushjain1144** [comment]
Ah I see, I forgot to update the docstring, probably that's what confused you. The [mask](https://github.com/ayushjain1144/odin/blob/cc60d01418b03f4d8db84429307ff6ae7d2622d3/odin/modeling/matcher.py#L129) here is basically of shape [num_queries, num_points] where num_points are all the points of the RGB-D pointcloud
